aten/src/ATen/native/sparse/SparseFactories.cpp

```diff
@@ -91,5 +95,49 @@ Tensor spdiags(
   return result_coo;
 }
 
+Tensor spdiags_backward(
+    const Tensor& grad_out,
+    const Tensor& offsets,
+    IntArrayRef input_shape) {
+  auto offsets_1d = offsets.dim() == 0 ? offsets.unsqueeze(0) : offsets;
+
+  auto n_diag = input_shape.size() == 2 ? input_shape[0] : 1;
+  auto n_col_in = input_shape.size() == 2 ? input_shape[1] : input_shape[0];
+  AT_ASSERT(grad_out.dim() == 2);
+  AT_ASSERT(offsets_1d.size(0) == n_diag);
+  auto grad_in_options = grad_out.options().layout(Layout::Strided);
+  // zeros since we are only going to se the non-zero elements
+  Tensor grad_in = at::zeros({n_diag, n_col_in}, grad_in_options);
+  auto grad_out_coo = grad_out.layout() == Layout::Sparse
+      ? grad_out.coalesce()
+      : grad_out.to_sparse();
```

Comment:
Could `grad_out` be a non-sparse Tensor?

Reply:
`grad_out` will always be a sparse tensor, but may have `sparse_csr`, or `sparse_csc` layout. It should never be a dense tensor though as `spdiags` will always output a sparse tensor.